docs/google/engine.md

```diff
@@ -273,4 +273,41 @@ historical_results = historical_job.results()
 ```
 
 If you did not save the ids, you can still find them from your
-job using the [Cloud Console](https://console.cloud.google.com/quantum/jobs).
+job using the [Cloud Console](https://console.cloud.google.com/quantum/jobs) or
+by using `cirq.google.Engine.list_jobs()` or `cirq.google.Engine.list_programs()`.
+You can search in all the jobs within your project using filtering criteria on
+creation time, execution state and labels.  
+
+```python
+from cirq.google.engine.client.quantum import enums
+
+# Initialize the engine object
+engine = cirq.google.Engine(project_id='YOUR_PROJECT_ID')
+
+# List all the jobs on the project since 2020/09/20 that succeeded:
+jobs = engine.list_jobs(created_after=datetime.date(2020,9,20),
+                        execution_states=[enums.ExecutionStatus.State.SUCCESS])
+for j in jobs:
+   print(j.job_id, j.status(), j.create_time())
+``` 
+
+Alternatively, you can list programs by creation time and labels, and then list 
```

Comment:
Nit: this is a little difficult to parse. Is it worth explaining _how_ they retrieve the program, or could you just say something like "With an existing EngineProgram object, you can list any jobs that were run using that program."?